ruff_lsp/server.py

```diff
@@ -1013,8 +1013,7 @@ def _update_workspace_settings(settings: list[WorkspaceSettings]) -> None:
             }
 
 
-def _get_document_key(document: workspace.TextDocument) -> str | None:
-    document_workspace = Path(document.path)
+def _get_document_key(document_workspace: Path) -> str | None:
```

Comment:
Can we not pass a `Document` in here and then call `Path` on it internally? What's the motivation for passing in `Path`?

Reply:
So, for notebook we have a different type `NotebookDocument`. The motivation being that the event handler will pass in either the notebook or file path along with the source code to the `_lint_document_impl`.

We could create a type `NotebookOrTextDocument` which would abstract out the `path` and `source` field since that's the only thing we use. Or, we could use a `Protocol`.

Reply:
I'm actually thinking of using `str` here instead and use `Path` whenever required instead of instantiating it early on.

Reply:
I'm actually thinking of using `str` here instead and use `Path` whenever required instead of instantiating it early on.

Reply:
Got it, what you have here seems reasonable.